### ISSUE TITLE
RawArrays do not work from within Dictionaries

### ISSUE THREAD

**Issue by acalc** [godotengine/godot#7073]
**Operating system or device - Godot version:**
OS: OSX 10.11.6
Godot: 2.1.stable.official

**Issue description** (what happened, and what was expected):
Referencing RawArrays from within Dictionaries and trying to manipulate them doesn't seem to work (and no warning/error messages are produced in the editor).

**Steps to reproduce:**
```gdscript
var d = { ra = RawArray() }
d["ra"].append_array("testing".to_utf8())
printt("result:", d["ra"].get_string_from_utf8()) 
# "result:    <blank>"
# expected "result:    testing"
```

append(), size(), etc don't seem to work, either (size() = 0). However, if you grab the RawArray ref from the dictionary and store it in a variable, then operations will work again. (You could then re-store the RawArray back in the Dictionary.)

```gdscript
# ...
var ra = d["ra"]
ra.append_array("this should work".to_utf8())
d["ra"] = ra
printt("result:", d["ra"].get_string_from_utf8())
# "result:    this should work"
```

If RawArrays can't work from Dictionaries for some reason, I suppose there should at least be some sort of error message in the debugger console.

**Comment by bojidar-bg** [comment]
I guess it comes from https://github.com/godotengine/godot/issues/5277

**Comment by Zylann** [comment]
Isn't it because RawArrays are passed by value rather than reference? (They don't inherit anything and are tagged as built-in types)

**Comment by bojidar-bg** [comment]
@Zylann Yeah, sort-of. RawArrays and Arrays and Dictionaries, when created using their respective constructors would be in COW mode, and not in "reference" mode. This would be fixed in 3.0, as it would break compatibility in some cases.

**Comment by bojidar-bg** [comment]
Closing as duplicate of #5277 as per above comments.
